doc/tutorials/enable_s5.rst

```diff
@@ -31,135 +31,202 @@ The diagram below shows the overall architecture:
+   ``The user VM is not allowed to trigger system shutdown``
+
+Initiate a System S5 within the Service VM
+==========================================
+
+On the Service VM side, it uses the ``s5_trigger_linux.py`` to trigger the system S5 flow. Then,
+the Lifecycle Manager in service VM sends a ``poweroff_cmd`` request to the lifecycle manager in each
+User VM through the vUART channel. If the User VM receives this request, it will send an ``ack_poweroff``
+to the lifecycle manager in Service VM. It is the Service VM's responsibility to check whether the
+User VMs shut down successfully or not, and to decide when to shut the Service VM itself down.
+
+.. note:: Service VM is always allowed to trigger system S5 by default.
 
 .. _enable_s5:
 
 Enable S5
 *********
 
-The procedure for enabling S5 is specific to the particular OS:
-
-* For Linux (LaaG) or Windows (WaaG), include these lines in the launch script:
-
-  .. code-block:: bash
+1. Configure communication vUART for Service VM and User VMs:
+
+   Add these lines in the hypervisor scenario XML file manually:
+
+   Example::
+
+      /* VM0 */
+      <vm_type>SERVICE_VM</vm_type>
+      ...
+      <legacy_vuart id="1">
+         <type>VUART_LEGACY_PIO</type>
+         <base>CONFIG_COM_BASE</base>
+         <irq>0</irq>
+         <target_vm_id>1</target_vm_id>
+         <target_uart_id>1</target_uart_id>
+      </legacy_vuart>
+      <legacy_vuart id="2">
+        <type>VUART_LEGACY_PIO</type>
+        <base>CONFIG_COM_BASE</base>
+        <irq>0</irq>
+        <target_vm_id>2</target_vm_id>
+        <target_uart_id>2</target_uart_id>
+      </legacy_vuart>
+      ...
+      /* VM1 */
+      <vm_type>POST_STD_VM</vm_type>
+      ...
+      <legacy_vuart id="1">
+         <type>VUART_LEGACY_PIO</type>
+         <base>COM2_BASE</base>
+         <irq>COM2_IRQ</irq>
+         <target_vm_id>0</target_vm_id>
+         <target_uart_id>1</target_uart_id>
+      </legacy_vuart>
+      ...
+      /* VM2 */
+      <vm_type>POST_STD_VM</vm_type>
+      ...
+      <legacy_vuart id="1">
+         <type>VUART_LEGACY_PIO</type>
+         <base>INVALID_COM_BASE</base>
+         <irq>COM2_IRQ</irq>
+         <target_vm_id>0</target_vm_id>
+         <target_uart_id>2</target_uart_id>
+      </legacy_vuart>
+      <legacy_vuart id="2">
+         <type>VUART_LEGACY_PIO</type>
+         <base>COM2_BASE</base>
+         <irq>COM2_IRQ</irq>
+         <target_vm_id>0</target_vm_id>
+         <target_uart_id>2</target_uart_id>
+      </legacy_vuart>
+      ...
+      /* VM3 */
+      ...
+
+  .. note:: These vUART is emulated in the hypervisor; expose the node as ``/dev/ttySn``.
+     For the User VM with the minimal VM ID, the communication vUART id should be 1.
+     For other User VMs, the vUART (id is 1) shoulbe be configured as invalid, the communication
+     vUART id should be 2 or others.
+  .. note:: serial.conf file will be generated for Service VM if above communication vUARTs
```

Comment:
What is in this `serial.conf` file?

Reply:
serial.conf is to help service VM to configure vUART emulated by hypervisor. It is generated during ACRN hypervisor build.